caddyhttp/fastcgi/fastcgi.go

```diff
@@ -100,6 +102,20 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 			case "OPTIONS":
 				resp, err = fcgiBackend.Options(env)
 			default:
+				// since some chrome releases or quic clients are not sending the Content-Length
+				// in the request headers, caddy should re-calculate content-length by the body and fix it.
+				if contentLength == 0 {
+					var body []byte
+					if r.Body != nil {
+						body, _ = ioutil.ReadAll(r.Body)
```

Comment:
I know it will create some problems with the memory management. even though it is not the most efficient way and also not often the case, I just noticed my stable chrome release does not send Content-Length header even though there is content inside the request so I just tried it with that way. (ye even if it's a special case, that's not good way though.) Well, I totally agree that content-length header should be included in every POST requests but It used "should" instead of "must", see content below.

```
A user agent SHOULD send a Content-Length in a request message when
   no Transfer-Encoding is sent and the request method defines a meaning
   for an enclosed payload body.  For example, a Content-Length header
   field is normally sent in a POST request even when the value is 0
   (indicating an empty payload body).  A user agent SHOULD NOT send a
   Content-Length header field when the request message does not contain
   a payload body and the method semantics do not anticipate such a
   body.
```

I haven't checked that it's working when only Transfer-Encoding is given, I think that caddy can (and should) handle this particular matter because it's impossible to ensure that clients behave as intended. or at least it could be better to send 411 Length Required if Content-Length is missing.